cdap-app-fabric-tests/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowStatsSLAHttpHandlerTest.java

```diff
@@ -63,13 +65,30 @@
   private static MetricStore metricStore;
   private static Store store;
 
+  private int sourceId;
+
   @BeforeClass
   public static void beforeClass() throws Throwable {
     AppFabricTestBase.beforeClass();
     store = getInjector().getInstance(DefaultStore.class);
     metricStore = getInjector().getInstance(MetricStore.class);
   }
 
+  private void setStartAndRunning(final ProgramId id, final String pid, final long startTime) {
+    setStartAndRunning(id, pid, startTime, ImmutableMap.<String, String>of(),
+                       ImmutableMap.<String, String>of());
+
+  }
+
+
+  private void setStartAndRunning(final ProgramId id, final String pid, final long startTime,
+                                  final Map<String, String> runtimeArgs,
+                                  final Map<String, String> systemArgs) {
+    store.setStart(id, pid, startTime, null, runtimeArgs, systemArgs,
+                   ByteBuffer.allocate(++sourceId).array());
+    store.setRunning(id, pid, startTime + 1, null, ByteBuffer.allocate(++sourceId).array());
```

Comment:
ByteBuffer.allocate(++sourceId).array() ? Is that really what you want? Create a byte buffer with capacity of sourceId, and then get the backing array? That array will not be exactly as long as a messageId. and it will be filled with zeros always. 

If I understand correctly, what you really need is to generate message Ids with increasing publish time. You can do that as follows:
```
static long sourceId = System.currentTimeMillis();
static byte[] createSourceMessageId() {
  byte[] buffer = new byte[MessageId.RAW_ID_SIZE];
  messageId.putRaw(sourceId++, 0, 0, 0, buffer, 0);
  return buffer;
}
```
By the way, you should never use array() on a ByteBuffer. It is easily perceived as returning the byte[] representation of the ByteBuffer, but that is not the case.